src/host/sparse.jl

```diff
@@ -0,0 +1,960 @@
+using LinearAlgebra
+using LinearAlgebra: BlasFloat
+
+abstract type AbstractGPUSparseArray{Tv, Ti, N} <: AbstractSparseArray{Tv, Ti, N} end
+const AbstractGPUSparseVector{Tv, Ti} = AbstractGPUSparseArray{Tv, Ti, 1}
+const AbstractGPUSparseMatrix{Tv, Ti} = AbstractGPUSparseArray{Tv, Ti, 2}
+
+abstract type AbstractGPUSparseMatrixCSC{Tv, Ti} <: AbstractGPUSparseArray{Tv, Ti, 2} end
+abstract type AbstractGPUSparseMatrixCSR{Tv, Ti} <: AbstractGPUSparseArray{Tv, Ti, 2} end
+abstract type AbstractGPUSparseMatrixCOO{Tv, Ti} <: AbstractGPUSparseArray{Tv, Ti, 2} end
+abstract type AbstractGPUSparseMatrixBSR{Tv, Ti} <: AbstractGPUSparseArray{Tv, Ti, 2} end
+
+const AbstractGPUSparseVecOrMat = Union{AbstractGPUSparseVector,AbstractGPUSparseMatrix}
+
+SparseArrays.nnz(g::T) where {T<:AbstractGPUSparseArray} = g.nnz
+SparseArrays.nonzeros(g::T) where {T<:AbstractGPUSparseArray} = g.nzVal
+
+SparseArrays.nonzeroinds(g::T) where {T<:AbstractGPUSparseVector} = g.iPtr
+SparseArrays.rowvals(g::T) where {T<:AbstractGPUSparseVector} = nonzeroinds(g)
+
+SparseArrays.rowvals(g::AbstractGPUSparseMatrixCSC) = g.rowVal
+SparseArrays.getcolptr(S::AbstractGPUSparseMatrixCSC) = S.colPtr
+
+Base.convert(T::Type{<:AbstractGPUSparseArray}, m::AbstractArray) = m isa T ? m : T(m)
+
+_dense_array_type(sa::SparseVector)     = SparseVector
+_dense_array_type(::Type{SparseVector}) = SparseVector
+_sparse_array_type(sa::SparseVector) = SparseVector
+_dense_vector_type(sa::AbstractSparseArray) = Vector
+_dense_vector_type(sa::AbstractArray)       = Vector
+_dense_vector_type(::Type{<:AbstractSparseArray}) = Vector
+_dense_vector_type(::Type{<:AbstractArray})       = Vector
+_dense_array_type(sa::SparseMatrixCSC)     = SparseMatrixCSC
+_dense_array_type(::Type{SparseMatrixCSC}) = SparseMatrixCSC
+_sparse_array_type(sa::SparseMatrixCSC)    = SparseMatrixCSC
+
+function _sparse_array_type(sa::AbstractGPUSparseArray) end
+function _dense_array_type(sa::AbstractGPUSparseArray) end
+function _coo_type(sa::AbstractGPUSparseArray) end
+_coo_type(::SA) where {SA<:AbstractGPUSparseMatrixCSC} = SA
```

Comment:
Why the `_` prefix? That signals to me these are internal methods, while AFAIU they should be implemented by back-ends.